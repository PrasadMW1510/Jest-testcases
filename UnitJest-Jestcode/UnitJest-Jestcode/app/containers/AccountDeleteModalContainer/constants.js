/*
 *
 * AccountDeleteModalContainer constants
 *
 */

export const POST_ACCOUNT_DELETE_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST';
export const POST_ACCOUNT_DELETE_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_FAILURE';
export const POST_ACCOUNT_DELETE_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_DELETE_REQUEST_SUCCESS';
export const POST_ACCOUNT_DELETE_MIA_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_MIA_DELETE_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST';
export const POST_ACCOUNT_UNENROLL_REQUEST_FAILURE =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_FAILURE';
export const POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS =
  'app/AccountDeleteModalContainer/POST_ACCOUNT_UNENROLL_REQUEST_SUCCESS';
