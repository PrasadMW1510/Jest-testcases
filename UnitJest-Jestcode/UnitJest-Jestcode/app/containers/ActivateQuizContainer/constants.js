/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
