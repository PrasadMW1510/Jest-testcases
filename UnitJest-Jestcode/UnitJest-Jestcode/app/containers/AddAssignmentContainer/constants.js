/*
 *
 * AddAssignmentContainer constants
 *
 */

export const DEFAULT_ACTION = 'app/AddAssignmentContainer/DEFAULT_ACTION';
export const GET_STUDENTDETAIL_SUCCESS = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_SUCCESS';
export const GET_STUDENTDETAIL_FAILURE = 'app/AddAssignmentContainer/GET_STUDENTDETAIL_FAILURE';
export const GET_STUDENT_DETAILS = 'app/AddAssignmentContainer/GET_STUDENT_DETAILS';
export const POST_NEW_ASSIGNMENT = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT';
export const POST_NEW_ASSIGNMENT_SUCCESS = 'app/AddAssignmentContainer/POST_NEW_ASSIGNMENT_SUCCESS';
