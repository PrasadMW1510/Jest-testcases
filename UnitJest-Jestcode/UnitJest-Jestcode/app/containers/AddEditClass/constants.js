/*
 *
 * AddEditClass constants
 *
 */

export const SAVE_CLASS_REQUEST = 'AddEditClass/save_class_request';
export const SAVE_CLASS_REQUEST_SUCCESS = 'AddEditClass/save_class_request_success';
export const SAVE_CLASS_REQUEST_FAILURE = 'AddEditClass/save_class_request_failure';
export const SAVE_CLASS_MIA_REQUEST = 'AddEditClass/save_class_mia_request';
export const SAVE_CLASS_MIA_REQUEST_SUCCESS = 'AddEditClass/save_class_mia_request_success';
export const SAVE_CLASS_MIA_REQUEST_FAILURE = 'AddEditClass/save_class_mia_request_failure';

export const INITIALIZE_CLASS_FORM_REQUEST = 'AddEditClass/initialize_class_form_request';
export const INITIALIZE_CLASS_FORM_REQUEST_SUCCESS =
  'AddEditClass/initialize_class_form_request_success';
export const INITIALIZE_CLASS_FORM_REQUEST_FAILURE =
  'AddEditClass/initialize_class_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_TEACHERS = 'meta_data_teachers';
export const META_DATA_STUDENTS = 'meta_data_students';
export const FIELD_SELECTED_STUDENTS = 'rosterStudents';
export const FORM_CLASS_PROFILE = 'form_class_profile';
export const FORM_TITLE_ADD = 'Add a Class';
export const FORM_TITLE_EDIT = 'Edit Class Profile';
