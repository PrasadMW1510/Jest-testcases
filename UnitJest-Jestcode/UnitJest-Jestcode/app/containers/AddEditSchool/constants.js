/*
 *
 * AddEditSchool constants
 *
 */

export const FORM_SCHOOL_PROFILE = 'form_school_profile';
export const FORM_TITLE_ADD = 'Add a School';
export const FORM_TITLE_EDIT = 'Edit School Profile';

// Actions

export const META_DATA_GRADES = 'meta_data_grades';

export const GRADE_LIST_FOR_DISTRICT = 'AddEditSchool/grade_list_for_district';
export const GET_META_DATA_REQUEST = 'AddEditSchool/get_meta_data_request';
export const GET_META_DATA_REQUEST_SUCCESS = 'AddEditSchool/get_meta_data_request_success';
export const GET_META_DATA_REQUEST_FAILURE = 'AddEditSchool/get_meta_data_request_failure';

export const INITIALIZE_SCHOOL_FORM_REQUEST = 'AddEditSchool/initialize_school_form_request';
export const INITIALIZE_SCHOOL_FORM_REQUEST_SUCCESS =
  'AddEditSchool/initialize_school_form_request_success';
export const INITIALIZE_SCHOOL_FORM_REQUEST_FAILURE =
  'AddEditSchool/initialize_school_form_request_failure';

export const SAVE_SCHOOL_REQUEST = 'AddEditSchool/save_school_request';
export const SAVE_SCHOOL_REQUEST_SUCCESS = 'AddEditSchool/save_school_request_success';
export const SAVE_SCHOOL_REQUEST_FAILURE = 'AddEditSchool/save_school_request_failure';

export const SAVE_SCHOOL_MIA_REQUEST = 'AddEditSchool/save_school_mia_request';
