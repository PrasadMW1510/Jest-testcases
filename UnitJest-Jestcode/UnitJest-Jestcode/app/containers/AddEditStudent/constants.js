/*
 *
 * AddEditStudent constants
 *
 */

export const SAVE_STUDENT_REQUEST = 'AddEditStudent/save_student_request';
export const SAVE_STUDENT_REQUEST_SUCCESS = 'AddEditStudent/save_student_request_success';
export const SAVE_STUDENT_REQUEST_FAILURE = 'AddEditStudent/save_student_request_failure';
export const FORM_STUDENT_PROFILE = 'FORM_STUDENT_PROFILE';
export const INITIALIZE_STUDENT_FORM_REQUEST = 'AddEditStudent/initialize_student_form_request';
export const INITIALIZE_STUDENT_FORM_REQUEST_SUCCESS =
  'AddEditStudent/initialize_student_form_request_success';
export const INITIALIZE_STUDENT_FORM_REQUEST_FAILURE =
  'AddEditStudent/initialize_student_form_request_failure';
export const META_DATA_GRADES = 'meta_data_grades';
export const META_DATA_CLASSES = 'meta_data_classes';
export const META_DATA_GROUPS = 'meta_data_groups';
export const META_DATA_PASSWORD_CONFIG = 'meta_data_password_config';
export const FORM_TITLE_ADD = 'Add a Student';
export const FORM_TITLE_EDIT = 'Edit Student Profile';
