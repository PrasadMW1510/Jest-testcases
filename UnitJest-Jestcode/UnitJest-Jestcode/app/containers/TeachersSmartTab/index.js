/**
 *
 * TeachersSmartTab
 *
 */

export { default } from './TeachersSmartTab';
