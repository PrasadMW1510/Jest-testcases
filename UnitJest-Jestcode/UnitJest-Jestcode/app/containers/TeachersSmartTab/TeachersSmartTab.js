/**
 *
 * TeachersSmartTab
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import SmartBarTab from 'components/SmartBarTab/SmartBarTab';
import {
  makeSelectTeachersData,
  makeSelectLoginData,
  makeSelectTeacherExpandCollapseStatus,
} from 'containers/App/selectors';
import {
  makeSelectClickedTeacherId,
  makeSelectClickedGradeData,
  makeSelectClickedSchoolData,
  makeSelectCohortType,
  makeSelectClickedGradeId,
  makeSelectClickedSchoolId,
  makeSelectTeacherId,
} from 'containers/SmartBarContainer/selectors';
import {
  teacherSelection,
  activeSelectedSchool,
  activeSelectedGrade,
  activeSelectedTeacher,
  activeSelectedClass,
  activeSelectedGroup,
  activeSelectedStudent,
  smartbarSelectedUpdateData,
} from 'containers/SmartBarContainer/actions';
import { COHORT_TYPE } from 'containers/App/constants';

export class TeachersSmartTab extends React.Component {
  componentWillReceiveProps(nextProps) {
    if (
      nextProps.selectedTeacherId === nextProps.smartBarSelectedTeacherId &&
      nextProps.selectedTeacherId !== '' &&
      nextProps.selectedCohortType === COHORT_TYPE.Teacher
    ) {
      this.props.activeSelectedSchool(nextProps.clickedSchoolId);
      this.props.activeSelectedGrade(nextProps.clickedGradeId);
      this.props.activeSelectedTeacher(nextProps.smartBarSelectedTeacherId);
      this.props.activeSelectedClass('');
      this.props.activeSelectedGroup('');
      this.props.activeSelectedStudent('');
    }
  }

  getTeachers = () => {
    if (!this.props.teachers) {
      return null;
    }

    return this.props.teachers
      .map(item => ({
        text: `${item.getIn(['last_name', 0])}, ${item.getIn(['first_name', 0])}`,
        id: item.getIn(['user_id', 0]),
      }))
      .toJS();
  };

  getForName = () => {
    if (this.props.selectedGradeData) {
      return this.props.selectedGradeData.getIn(['full_name', 0]);
    } else if (this.props.selectedSchoolData) {
      return this.props.selectedSchoolData.getIn(['name', 0]);
    }

    return `${this.props.login.getIn(['first_name', 0])} ${this.props.login.getIn([
      'last_name',
      0,
    ])}`;
  };

  shouldDisplay = () => {
    const userOrg = this.props.login.getIn(['user_org', 0]);
    return userOrg === COHORT_TYPE.District || userOrg === COHORT_TYPE.School;
  };

  render() {
    if (!this.shouldDisplay()) {
      return null;
    }

    return (
      <SmartBarTab
        title="Teachers"
        items={this.getTeachers()}
        forName={this.getForName()}
        onItemClick={this.props.teacherSelection}
        selectedItemId={this.props.selectedTeacherId}
        defaultChecked={this.props.teacherStatus}
        isTabActive={this.props.selectedCohortType === COHORT_TYPE.Teacher}
        smartbarSelectedUpdateData={this.props.smartbarSelectedUpdateData}
      />
    );
  }
}

TeachersSmartTab.propTypes = {
  teachers: PropTypes.object,
  selectedTeacherId: PropTypes.string,
  login: PropTypes.object.isRequired,
  teacherSelection: PropTypes.func.isRequired,
  selectedGradeData: PropTypes.object,
  selectedSchoolData: PropTypes.object,
  selectedCohortType: PropTypes.string,
  activeSelectedSchool: PropTypes.func.isRequired,
  activeSelectedGrade: PropTypes.func.isRequired,
  activeSelectedTeacher: PropTypes.func.isRequired,
  activeSelectedClass: PropTypes.func.isRequired,
  activeSelectedGroup: PropTypes.func.isRequired,
  activeSelectedStudent: PropTypes.func.isRequired,
  clickedGradeId: PropTypes.string,
  clickedSchoolId: PropTypes.string,
  smartBarSelectedTeacherId: PropTypes.string,
  smartbarSelectedUpdateData: PropTypes.func.isRequired,
  teacherStatus: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  teachers: makeSelectTeachersData(),
  selectedTeacherId: makeSelectClickedTeacherId(),
  selectedGradeData: makeSelectClickedGradeData(),
  selectedSchoolData: makeSelectClickedSchoolData(),
  smartBarSelectedTeacherId: makeSelectTeacherId(),
  clickedGradeId: makeSelectClickedGradeId(),
  clickedSchoolId: makeSelectClickedSchoolId(),
  selectedCohortType: makeSelectCohortType(),
  login: makeSelectLoginData(),
  teacherStatus: makeSelectTeacherExpandCollapseStatus(),
});

const withConnect = connect(mapStateToProps, {
  teacherSelection,
  activeSelectedSchool,
  activeSelectedGrade,
  activeSelectedTeacher,
  activeSelectedClass,
  activeSelectedGroup,
  activeSelectedStudent,
  smartbarSelectedUpdateData,
});

export default compose(withConnect)(TeachersSmartTab);
