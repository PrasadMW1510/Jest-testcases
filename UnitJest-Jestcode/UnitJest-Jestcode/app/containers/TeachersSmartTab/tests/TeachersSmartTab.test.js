import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { fromJS } from 'immutable';
import SmartBarTab from 'components/SmartBarTab';
import { COHORT_TYPE } from 'containers/App/constants';
import { TeachersSmartTab } from '../TeachersSmartTab';

describe('<TeachersSmartTab />', () => {
  let wrapper = null;
  let mockedGlobalState = null;
  let mockedLoginState = null;
  let mockedTeacherSelection = null;
  let mockedSelections = null;
  let mockActiveSelectedSchool = null;
  let mockActiveSelectedGrade = null;
  let mockActiveSelectedTeacher = null;
  let mockActiveSelectedClass = null;
  let mockActiveSelectedGroup = null;
  let mockActiveSelectedStudent = null;
  let mockClickedGradeId = null;
  let mockClickedSchoolId = null;
  let mockSmartBarSelectedTeacherId = null;
  let mockSmartbarSelectedUpdateData = null;
  let mockTeacherStatus = null;

  beforeEach(() => {
    mockedGlobalState = fromJS([
      { user_id: ['teacher1'], first_name: ['john'], last_name: ['smith'] },
      { user_id: ['teacher2'], first_name: ['jane'], last_name: ['doe'] },
    ]);
    mockedLoginState = fromJS({
      user_org: [COHORT_TYPE.District],
      first_name: ['Sam'],
      last_name: ['Smith'],
    });

    mockActiveSelectedSchool = jest.fn();
    mockActiveSelectedGrade = jest.fn();
    mockActiveSelectedTeacher = jest.fn();
    mockActiveSelectedClass = jest.fn();
    mockActiveSelectedGroup = jest.fn();
    mockActiveSelectedStudent = jest.fn();
    mockSmartbarSelectedUpdateData = jest.fn();
    mockClickedSchoolId = 'randomSchool';
    mockClickedGradeId = 'randomGrade';
    mockSmartBarSelectedTeacherId = '';

    mockedTeacherSelection = jest.fn();
    mockedSelections = 'sadfasfas';
    mockTeacherStatus = true;
    wrapper = shallow(
      <TeachersSmartTab
        selectedTeacherId={mockedSelections}
        teachers={mockedGlobalState}
        teacherSelection={mockedTeacherSelection}
        login={mockedLoginState}
        activeSelectedSchool={mockActiveSelectedSchool}
        activeSelectedGrade={mockActiveSelectedGrade}
        activeSelectedTeacher={mockActiveSelectedTeacher}
        activeSelectedClass={mockActiveSelectedClass}
        activeSelectedGroup={mockActiveSelectedGroup}
        activeSelectedStudent={mockActiveSelectedStudent}
        clickedSchoolId={mockClickedSchoolId}
        clickedGradeId={mockClickedGradeId}
        smartBarSelectedTeacherId={mockSmartBarSelectedTeacherId}
        smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
        teacherStatus={mockTeacherStatus}
      />
    );
  });

  it('Expect to have a SmartBarTab', () => {
    expect(wrapper.find(SmartBarTab).exists()).toBeTruthy();
  });

  it('verify with the selected value is the same', () => {
    wrapper.setProps({ smartBarSelectedTeacherId: 'sadfasfas', selectedCohortType: 'Teacher' });
    expect(mockActiveSelectedTeacher).toHaveBeenCalled();
  });

  it('verify with the value is not equal', () => {
    wrapper.setProps({ smartBarSelectedTeacherId: '' });
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Expect to have a SmartBarTab for school users', () => {
    mockedLoginState = mockedLoginState.setIn(['user_org', 0], COHORT_TYPE.School);
    wrapper = shallow(
      <TeachersSmartTab
        selectedTeacherId={mockedSelections}
        teachers={mockedGlobalState}
        teacherSelection={mockedTeacherSelection}
        login={mockedLoginState}
        activeSelectedSchool={mockActiveSelectedSchool}
        activeSelectedGrade={mockActiveSelectedGrade}
        activeSelectedTeacher={mockActiveSelectedTeacher}
        activeSelectedClass={mockActiveSelectedClass}
        activeSelectedGroup={mockActiveSelectedGroup}
        activeSelectedStudent={mockActiveSelectedStudent}
        smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
        teacherStatus={mockTeacherStatus}
      />
    );
    expect(wrapper.find(SmartBarTab).exists()).toBeTruthy();
  });

  it('Expect to not display for teachers', () => {
    mockedLoginState = mockedLoginState.setIn(['user_org', 0], null);
    wrapper = shallow(
      <TeachersSmartTab
        selectedTeacherId={mockedSelections}
        teachers={mockedGlobalState}
        teacherSelection={mockedTeacherSelection}
        login={mockedLoginState}
        activeSelectedSchool={mockActiveSelectedSchool}
        activeSelectedGrade={mockActiveSelectedGrade}
        activeSelectedTeacher={mockActiveSelectedTeacher}
        activeSelectedClass={mockActiveSelectedClass}
        activeSelectedGroup={mockActiveSelectedGroup}
        activeSelectedStudent={mockActiveSelectedStudent}
        smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
        teacherStatus={mockTeacherStatus}
      />
    );
    expect(wrapper.find(SmartBarTab).exists()).toBeFalsy();
  });

  it('Expect to have a SmartBarTab with the title Classes', () => {
    expect(wrapper.find(SmartBarTab).prop('title')).toEqual('Teachers');
  });

  it('Expect to pass mocked global state classes to the SmartBarTab', () => {
    expect(wrapper.find(SmartBarTab).prop('items').size).toEqual(mockedGlobalState.length);
  });

  it('Expect to handle no teachers', () => {
    wrapper = shallow(
      <TeachersSmartTab
        selectedTeacherId={mockedSelections}
        teacherSelection={mockedTeacherSelection}
        login={mockedLoginState}
        activeSelectedSchool={mockActiveSelectedSchool}
        activeSelectedGrade={mockActiveSelectedGrade}
        activeSelectedTeacher={mockActiveSelectedTeacher}
        activeSelectedClass={mockActiveSelectedClass}
        activeSelectedGroup={mockActiveSelectedGroup}
        activeSelectedStudent={mockActiveSelectedStudent}
        smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
        teacherStatus={mockTeacherStatus}
      />
    );
    expect(wrapper.find(SmartBarTab).prop('items')).toBeFalsy();
  });

  it('Should pass dispatch to child element', () => {
    const smartBar = wrapper.find(SmartBarTab);
    smartBar.prop('onItemClick')();
    expect(mockedTeacherSelection).toHaveBeenCalled();
  });

  describe('For name', () => {
    let mockedGradeData = null;
    let mockedSchoolData = null;

    beforeEach(() => {
      mockedGradeData = fromJS({
        full_name: ['Kindergarten'],
      });

      mockedSchoolData = fromJS({
        name: ['School 1'],
      });
    });

    it('Should display the logged in user name if no other data is present', () => {
      expect(wrapper.find(SmartBarTab).prop('forName')).toMatchSnapshot();
    });

    it('Should display the selected grade name if present', () => {
      wrapper = shallow(
        <TeachersSmartTab
          selectedGradeData={mockedGradeData}
          selectedTeacherId={mockedSelections}
          teachers={mockedGlobalState}
          teacherSelection={mockedTeacherSelection}
          login={mockedLoginState}
          activeSelectedSchool={mockActiveSelectedSchool}
          activeSelectedGrade={mockActiveSelectedGrade}
          activeSelectedTeacher={mockActiveSelectedTeacher}
          activeSelectedClass={mockActiveSelectedClass}
          activeSelectedGroup={mockActiveSelectedGroup}
          activeSelectedStudent={mockActiveSelectedStudent}
          smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
          teacherStatus={mockTeacherStatus}
        />
      );
      expect(wrapper.find(SmartBarTab).prop('forName')).toMatchSnapshot();
    });

    it('Should display the selected school name if present', () => {
      wrapper = shallow(
        <TeachersSmartTab
          selectedSchoolData={mockedSchoolData}
          selectedTeacherId={mockedSelections}
          teachers={mockedGlobalState}
          teacherSelection={mockedTeacherSelection}
          login={mockedLoginState}
          activeSelectedSchool={mockActiveSelectedSchool}
          activeSelectedGrade={mockActiveSelectedGrade}
          activeSelectedTeacher={mockActiveSelectedTeacher}
          activeSelectedClass={mockActiveSelectedClass}
          activeSelectedGroup={mockActiveSelectedGroup}
          activeSelectedStudent={mockActiveSelectedStudent}
          smartbarSelectedUpdateData={mockSmartbarSelectedUpdateData}
          teacherStatus={mockTeacherStatus}
        />
      );
      expect(wrapper.find(SmartBarTab).prop('forName')).toMatchSnapshot();
    });
  });
});
