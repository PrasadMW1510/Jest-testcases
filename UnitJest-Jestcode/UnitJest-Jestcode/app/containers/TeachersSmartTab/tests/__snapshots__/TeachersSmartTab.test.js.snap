// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeachersSmartTab /> For name Should display the logged in user name if no other data is present 1`] = `"Sam Smith"`;

exports[`<TeachersSmartTab /> For name Should display the selected grade name if present 1`] = `"Kindergarten"`;

exports[`<TeachersSmartTab /> For name Should display the selected school name if present 1`] = `"School 1"`;

exports[`<TeachersSmartTab /> verify with the value is not equal 1`] = `
<SmartBarTab
  defaultChecked={true}
  forName="Sam Smith"
  isTabActive={false}
  items={
    Array [
      Object {
        "id": "teacher1",
        "text": "smith, john",
      },
      Object {
        "id": "teacher2",
        "text": "doe, jane",
      },
    ]
  }
  onItemClick={[Function]}
  selectedItemId="sadfasfas"
  smartbarSelectedUpdateData={[Function]}
  title="Teachers"
/>
`;
